src/RevTracer.cc

```diff
@@ -0,0 +1,336 @@
+//
+// _RevTracer_cc_
+//
+// Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+// All Rights Reserved
+// contact@tactcomplabs.com
+//
+// See LICENSE in the top level directory for licensing details
+//
+//
+
+#include <sstream>
+#include <iomanip>
+#include <string>
+
+#include "../include/RevTracer.h"
+#include "RevTracer.h"
+
+using namespace SST::RevCPU;
+
+RevTracer::RevTracer(std::string Name, SST::Output *o)
+    : name(Name), pOutput(o), outputEnabled(false), lastPC(0), insn(0), 
+    traceSymbols(nullptr), startCycle(0), cycleLimit(0), traceCycles(0), 
+    disabled(0) {
+    
+    enableQ.resize(MAX_ENABLE_Q);
+    enableQ.assign(MAX_ENABLE_Q,0);
+    enableQindex = 0;
+
+    // Initialize NOP trace controls
+    uint32_t cmd_template = s2op.at(TRC_OP_DEFAULT);
+    for (unsigned i=0;i<NOP_COUNT;i++)
+        nops[i]= cmd_template | (i << TRC_OP_POS);
+
+    #if 0
+    if (std::getenv("REV_SPINNER")){
+        uint64_t spinner = 1;
+        std::cout << "spinner active" << std::endl;
+        while (spinner){
+            spinner++;
+            if (spinner % 10000000 == 0) // break here
+                std::cout << ".";
+        }
+        std::cout << std::endl;
+    }
+    #endif
+}
+
+RevTracer::~RevTracer()
+{
+    #ifdef REV_USE_SPIKE
+    if (diasm) delete diasm;
+    if (isaParser) delete isaParser;
+    #endif
+}
+
+int RevTracer::SetDisassembler(std::string machine)
+{
+    #ifdef REV_USE_SPIKE
+    try {
+        // TODO privelege level options
+        isaParser = new isa_parser_t(machine.c_str(),"MSU");
+        diasm = new disassembler_t(isaParser);
+    } catch (...) {
+        return 1;
+    }
+    return 0;
+    #else
+    return 1;
+    #endif
+}
+
+int RevTracer::SetTraceSymbols(std::map<uint64_t, std::string> *TraceSymbols)
+{
+    traceSymbols = TraceSymbols;
+    return 0;
+}
```

Comment:
Unless this is a part of a family of functions which return `int`, the return type should be `void`